torch/_inductor/codecache.py

```diff
@@ -379,6 +388,163 @@ def write_atomic(path: str, content: Union[str, bytes]) -> None:
     tmp_path.rename(path)
 
 
+def _reduce_fake_tensor(t):
+    """
+    See FxGraphCachePickler. Custom reducer to enable pickling FakeTensor.
+    """
+    return (TensorMetadataHolder, (_extract_tensor_metadata(t), t.device))
+
+
+@dataclasses.dataclass
+class TensorMetadataHolder:
+    tensor_metadata: TensorMetadata
+    device: torch.device
+
+
+def _reduce_typed_storage(s):
+    """
+    See FxGraphCachePickler. Custom reducer to enable pickling TypedStorage.
+    """
+    # The string repr seems to capture the relevant details for hashing, e.g.,
```

Comment:
For some of the benchmarks I tested, I saw TypedStorage objects in the torch.fx.GraphModule, and those were preventing cache hits when I straightforwardly pickle the whole object without playing this trick (because the _cdata field changes among runs). I don't _really_ know what a TypedStorage is, but from the docs sounds like some lower-level object under Tensor.

I could use some guidance here on the salient data to capture in order to hash safely. I don't _really_ like using the string representation, but to my naive eyes it seems to capture the relevant attributes of a TypedStorage, plus feels kinda future proof in case other fields are added in the future. Perhaps we could ignore the actual value, however (0.0 in the example I show here)??? In my testing, including the value didn't have an impact on cache hits for torchbench benchmarks.

Reply:
Where do you see typedstorage objects in the graphmodule? On the metadata?

Reply:
> I don't know when we would be using a Storage and not a Tensor in compilation 🤔 more information would be helpful here

@eellison, ok the instances of TypedStorage that I'm seeing _are_ in Tensors. I don't understand exactly what's going on, but I've discovered that much. Does it make sense that there would be Tensor objects in the graphmodule? I guess I naively assumed that every Tensor would be a FakeTensor. It seems like I can treat Tensors the same was as FakeTensors for the purposes of computing the hash key, but I think you also have a concern about that approach?

Reply:
Constants will appear as attributes of the graph module. Some of the constants are inlined directly into codegen, in which case we'd need to check for complete equality of values. The ones that are not inlined we'd just need to check metadata equality... Maybe this is better handled as a stacked pr / follow up, and bail for now when constants appear ? Up to you.